changelog.d/pr-8557

```diff
@@ -0,0 +1,20 @@
+synopsis: Add `--enable-semaphore` flag to enable interaction with GHC Job Server protocol
+packages: cabal-install
+prs: #8557
+
+description: {
+
+When cabal-install is passed the `--enable-semaphore` flag it will now act as a job server
+according to the GHC Jobserver Protocol.
+
+In particular this means that cabal-install will create a semaphore which it then
+passes to `./Setup build` (and hence `ghc`) which can be used by `ghc` in order to
+control how much paralellism it uses, coordinating with other simultaneously running
+processes.
+
+This feature requires ghc-9.8 in order to use, as this is the first version of GHC
```

Comment:
non-blocking: Shouldn't we always mention GHC 9.7? Because I fear that people may not find anything related to 9.8 if they look it up in the code base.

Reply:
I don't know what is best, it seems that most users won't know what ghc-9.7 is and won't be able to install it.

Reply:
Alright, let's leave it like this then. :)